### PR TITLE
JSON filters

### PR DESCRIPTION
**Pull request by jncarver**
#### What does this PR do?

This pull request adds filters for validating and parsing JSON strings (see #2). The validate function will return the original value without modification, while the parse function will parse it into an array, int, float, double, bool, or null, as returned by `json_decode`.

#### Checklist
- [X] Pull request contains a clear definition of changes
- [X] Tests (either unit, integration, or acceptance) written and passing
- [X] Relevant documentation produced and/or updated

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "license": "MIT",
     "require": {
         "php": "^7.0",
+        "ext-json": "*",
```
Comment (chadicus) [inline composer.json]: Although this is an edge case because most PHP instances will have json enabled. 
The best practice for this would be to add this as a dev requirement as it is required for unit tests, and add a `suggest` for `ext-json` in composer.json
```json
 "suggest": {
        "ext-json": "Required for Json filters"
    }
```

**Review by chadicus — CHANGES_REQUESTED** [review]
Code looks good, just need to update composer.json

File: composer.json (hunk 1 above)
Comment (jncarver) [reply composer.json]: Alright, I'll make that change.

**Review by chadicus — APPROVED** [review]
